proxy/pkg/zdmproxy/clusterconn.go

```diff
@@ -472,3 +477,20 @@ func (cc *ClusterConnector) sendAsyncRequest(
 
 	return err == nil
 }
+
+var heartBeatTime = time.Now()
+
+func (cc *ClusterConnector) sendHeartbeat() {
+	if time.Since(heartBeatTime) < 1*time.Minute {
+		return
+	}
+	heartBeatTime = time.Now()
+	optionsMsg := &message.Options{}
+	heartBeatFrame := frame.NewFrame(ccProtocolVersion, -1, optionsMsg)
```

Comment:
Using the static `ccProtocolVersion` const here causes some errors in the logs. It points to v3, but if the proxy has negotiated v4, it will complain.
@joao-r-reis Can I determine the version in use from here?

Reply:
`ccProtocolVersion` should be used on the control connection only. 
I believe the best way to get the protocol version here is to look at the startup frame that is stored on the client handler (`ch.startupRequest`) but in order to do that you will have to change that field to be thread safe because currently it is not.

Reply:
Thanks for the pointer, that seems to work in my initial tests (pushed the change to use the startupRequest in the `sendHeartbeat` function). I still need to add the mutex around that var though, that's coming next.

Reply:
Turned `startupRequest` into an atomic value.